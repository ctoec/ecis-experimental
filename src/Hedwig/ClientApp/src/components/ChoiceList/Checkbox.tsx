import React from 'react';

type CheckboxProps = {
	id: string;
	text: string;
	value: string;
	name?: string;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => any;
	selected?: boolean;
	className?: string;
	disabled?: boolean;
};

export default function Checkbox({
	id,
	text,
	value,
	name,
	onChange,
	selected,
	className,
	disabled,
}: CheckboxProps) {
	return (
		<div className={`usa-checkbox ${className}`}>
			<input
				className="usa-checkbox__input"
				id={id}
				type="checkbox"
				name={name || ''}
				value={value}
				defaultChecked={selected}
				onChange={onChange}
				disabled={disabled}
			/>
			<label className="usa-checkbox__label" htmlFor={id}>
				{text}
			</label>
		</div>
	);
}
