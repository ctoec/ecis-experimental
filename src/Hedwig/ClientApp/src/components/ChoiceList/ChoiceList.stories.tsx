import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { ChoiceList, FormStatusProps } from '..';

const onChange = action('onChange');
const options = [
	{
		text: 'Option one',
		value: 'one',
	},
	{
		text: 'Option two',
		value: 'two',
	},
];
const warning: FormStatusProps = {
	type: 'warning',
	message: 'These fields need your attention',
	id: 'checklist-warning',
};
const error: FormStatusProps = {
	type: 'error',
	message: 'These fields will block your progress',
	id: 'checklis-error',
};
const success: FormStatusProps = {
	type: 'success',
	message: 'These fields were validated woo',
	id: 'checklist-success',
};

storiesOf('ChoiceList', module)
	.add('Radio', () => {
		return (
			<ChoiceList
				type="radio"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				legend="Checklist items"
			/>
		);
	})
	.add('Radio with preselected option', () => {
		return (
			<ChoiceList
				type="radio"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				legend="Checklist items"
				selected={['one']}
			/>
		);
	})
	.add('Radio with error', () => {
		return (
			<ChoiceList
				type="radio"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				legend="Checklist items"
				status={error}
			/>
		);
	})
	.add('Checklist', () => {
		return (
			<ChoiceList
				type="check"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				legend="Checklist items"
			/>
		);
	})
	.add('Single checkbox', () => {
		return (
			<ChoiceList
				type="check"
				onChange={onChange}
				options={[options[0]]}
				id="storybook-checklist"
				legend="Checklist items"
			/>
		);
	})
	.add('Checklist with warning', () => {
		return (
			<ChoiceList
				type="check"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				legend="Checklist items"
				status={warning}
			/>
		);
	})
	.add('Select', () => {
		return (
			<ChoiceList
				label="Select"
				type="select"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
			/>
		);
	})
	.add('Select with success', () => {
		return (
			<ChoiceList
				label="Select"
				type="select"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				status={success}
			/>
		);
	})
	.add('Select with other', () => {
		return (
			<ChoiceList
				label="Select"
				type="select"
				onChange={onChange}
				options={options}
				id="storybook-checklist"
				otherInputLabel="Other choice"
			/>
		);
	});
