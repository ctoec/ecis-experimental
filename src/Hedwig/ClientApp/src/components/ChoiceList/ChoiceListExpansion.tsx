import React from 'react';

type ChoiceListExpansionProps = {
	showOnValue: string;
};

export const ChoiceListExpansion: React.FC<ChoiceListExpansionProps> = ({ children }) => {
	return <>{children}</>;
};
