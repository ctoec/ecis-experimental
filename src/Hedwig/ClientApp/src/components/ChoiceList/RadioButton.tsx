import React from 'react';

type RadioButtonProps = {
	text: string;
	value: string;
	name: string;
	onChange: (event: React.ChangeEvent<HTMLInputElement>) => any;
	selected?: boolean;
	disabled?: boolean;
	className?: string;
};

export default function RadioButton({
	text,
	value,
	name,
	onChange,
	selected,
	disabled,
	className,
}: RadioButtonProps) {
	return (
		<div className={`usa-radio ${className}`}>
			<input
				className="usa-radio__input"
				id={value}
				type="radio"
				name={name}
				value={value}
				checked={selected}
				disabled={!!disabled}
				onChange={onChange}
			/>
			<label className="usa-radio__label" htmlFor={value}>
				{text}
			</label>
		</div>
	);
}
