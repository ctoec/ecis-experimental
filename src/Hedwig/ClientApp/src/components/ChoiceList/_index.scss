.oec-choicelist-expansion {
	border-left: solid $theme-color-primary 4px;
	margin-left: 10px;
	padding-left: 1.3em;

	margin-bottom: 0.8em;

	> :first-child {
		padding-top: 0.7em;
	}

	> * {
		padding-bottom: 1em;
	}
}
