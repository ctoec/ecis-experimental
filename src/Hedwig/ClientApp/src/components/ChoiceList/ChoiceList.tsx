import React, { useState } from 'react';
import { TextInput, FormStatus, FormStatusProps, FieldSet } from '..';
import Checkbox from '../Checkbox/Checkbox';
import RadioButton from '../RadioButton/RadioButton';
import { ChoiceListExpansion } from './ChoiceListExpansion';

type Option = {
	text: string;
	value: string;
	name?: string;
};

type InternalOption = Option & {
	expansion?: JSX.Element;
};

export type HTMLChoiceElement = HTMLInputElement | HTMLSelectElement;

type InternalChoiceListProps = {
	options: Option[];
	id: string;
	name?: string;
	onChange: (
		e: React.ChangeEvent<HTMLChoiceElement>,
		selectedValues: string[],
		otherInput?: string
	) => any;
	defaultValue?: string[];
	status?: FormStatusProps;
	disabled?: boolean;
	optional?: boolean;
	className?: string;
	hint?: string;
	otherInputLabel?: string;
};

export type RadioOrChecklistProps = InternalChoiceListProps & {
	type: 'radio' | 'check';
	legend: string;
	horizontal?: boolean;
	label?: never;
	unselectedText?: never;
};

type DropdownProps = InternalChoiceListProps & {
	type: 'select';
	label: string;
	unselectedText?: string;
	horizontal?: never;
	legend?: never;
};

export type ChoiceListProps = RadioOrChecklistProps | DropdownProps;

/**
 * @deprecated
 */
export const ChoiceList: React.FC<ChoiceListProps> = ({
	type,
	options: inputOptions,
	id,
	name,
	onChange,
	defaultValue: inputSelected = [],
	legend,
	status,
	disabled,
	optional,
	className,
	hint,
	horizontal,
	otherInputLabel,
	label,
	unselectedText,
	children: expansionChildren,
}) => {
	const [selectedItems, updateSelection] = useState(inputSelected);
	const [otherInput, updateOtherInput] = useState<string | undefined>();

	let internalOptions = inputOptions as InternalOption[];
	const validTypesArray: boolean[] =
		React.Children.map(expansionChildren, (child) => {
			if (React.isValidElement(child)) {
				return !!child.type && (child.type as Function).name === ChoiceListExpansion.name;
			} else {
				throw new Error('Invalid element');
			}
		}) || [];
	const areValid = validTypesArray.reduce(
		(pendingAreValid, isValid) => pendingAreValid && isValid,
		true
	);
	if (!areValid) {
		throw new Error('One or more children are not of type ChoiceListExpansion');
	}
	const valueToExpansionChild = React.Children.toArray(expansionChildren).reduce<{
		[value: string]: JSX.Element;
	}>((acc, child) => {
		if (React.isValidElement(child)) {
			return {
				...acc,
				[child.props.showOnValue as string]: child,
			};
		} else {
			return acc;
		}
	}, {});
	internalOptions = inputOptions.map((option) => ({
		...option,
		expansion: valueToExpansionChild[option.value],
	}));

	const changeEvent = (event: React.ChangeEvent<HTMLChoiceElement>) => {
		const changedValue = event.target.value;
		if (event.target.type === 'text') {
			// If it's the other text input box
			updateOtherInput(changedValue);
			// Do whatever the dev wants done when things change
			onChange(event, selectedItems, changedValue);
			return;
		}
		let newSelectedItems: string[];
		if (selectedItems.includes(changedValue)) {
			// Uncheck a checkbox if it was already checked
			newSelectedItems = selectedItems.filter((v) => v !== changedValue);
		} else if (type === 'check') {
			// If it wasn't already selected and it's a checkbox, add it to whatever else is selected
			newSelectedItems = [changedValue, ...selectedItems];
		} else {
			// If it's a radio or dropdown, only this value is selected now
			newSelectedItems = [changedValue];
		}
		updateSelection(newSelectedItems);
		onChange(event, newSelectedItems, otherInput);
	};

	const options = [...internalOptions];
	if (otherInputLabel !== undefined) {
		options.push({
			text: otherInputLabel,
			value: 'other',
		});
	}

	const showotherInput = selectedItems.includes('other');
	let children: JSX.Element[] = [];

	switch (type) {
		case 'radio':
			if (selectedItems.length > 1) {
				throw new Error('Radio group can only have one selected value at a time.');
			}
			children = options.map((option) => {
				const expansion = option.expansion;
				return (
					<>
						<RadioButton
							{...option}
							name={option.name || name || ''}
							onChange={changeEvent}
							selected={selectedItems.includes(option.value)}
							disabled={disabled}
							// Including whether it's selected to force re-render... seems bad, but otherwise this option is not showing up as checked
							key={`${id}-${option.value}-${selectedItems.includes(option.value)}`}
						/>
						{expansion && selectedItems.includes(option.value) && (
							<div className="oec-choicelist-expansion">{expansion}</div>
						)}
					</>
				);
			});
			break;
		case 'check':
			children = options.map((option) => {
				const expansion = option.expansion;
				return (
					<>
						<Checkbox
							id={`${id}-${option.value}`}
							text={option.text}
							value={option.value}
							name={option.name || name || ''}
							onChange={changeEvent}
							defaultValue={selectedItems.includes(option.value)}
							disabled={disabled}
							key={`${id}-${option.value}`}
						/>
						{expansion && selectedItems.includes(option.value) && (
							<div className="oec-choicelist-expansion">{expansion}</div>
						)}
					</>
				);
			});
			break;
		case 'select':
			if (selectedItems.length > 1) {
				throw new Error('Dropdown can only have one selected value at a time.');
			}
			const optionElements = [
				<option key={`${id}-unselected`} value="">
					{unselectedText || '- Select -'}
				</option>,
			];
			options.forEach((option) => {
				optionElements.push(
					<option value={option.value} key={`${id}-${option.value}`}>
						{option.text}
					</option>
				);
			});
			children = [
				<select
					key={`${id}-select`}
					className={`usa-select${status ? ` usa-input--${status.type}` : ''}`}
					name={name || ''}
					id={id}
					onChange={changeEvent}
					disabled={disabled}
					value={selectedItems[0]}
					aria-required={!optional}
					// Using aria-required to avoid default Chrome behavior
					aria-describedby={status ? status.id : undefined}
					aria-invalid={status && status.type === 'error'}
				>
					{[...optionElements]}
				</select>,
			];
	}

	if (children.length === 1) {
		const singletonInput = (
			<div
				className={`usa-form-group${status ? ` usa-form-group--${status.type}` : ''} ${
					className || ''
				}`}
				key={`${id}-form-group`}
			>
				{hint && <span className="usa-hint text-italic">{hint}</span>}
				<div className={!label ? 'margin-top-3' : ''}></div>
				{label && (
					<label className={`usa-label${status ? ` usa-label--${status.type}` : ''}`} htmlFor={id}>
						{label}
					</label>
				)}
				{status && status.message && <FormStatus {...status} />}
				{[...children]}
				{options.map((option) => {
					const expansion = option.expansion;
					return (
						<React.Fragment key={`${option.value}-expansion`}>
							{expansion && selectedItems.includes(option.value) && (
								<div className="oec-choicelist-expansion">{option.expansion}</div>
							)}
						</React.Fragment>
					);
				})}
			</div>
		);
		if (!showotherInput) {
			return singletonInput;
		}
		children = [singletonInput];
	}

	return (
		<FieldSet
			// Will never actually be undefined
			legend={legend || label || ''}
			status={status}
			id={`${id}-fieldset`}
			aria-describedby={status ? status.id : undefined}
			className={className}
			childrenGroupClassName="margin-top-3"
			hint={hint}
			optional={optional}
		>
			<div className={horizontal ? 'grid-row flex-align-start grid-gap' : ''}>
				{[...children]}
				{showotherInput && otherInputLabel !== undefined && (
					<TextInput
						type="input"
						id={`${id}-other`}
						name={name || ''}
						label={otherInputLabel}
						onChange={changeEvent}
					/>
				)}
			</div>
		</FieldSet>
	);
};
