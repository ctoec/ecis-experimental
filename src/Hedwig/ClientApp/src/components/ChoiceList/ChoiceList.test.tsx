import React from 'react';
import { render } from '@testing-library/react';
import { ChoiceList } from './ChoiceList';
import { ChoiceListExpansion } from './ChoiceListExpansion';

function createChoiceList(type: 'select' | 'radio' | 'check', selected?: string[]) {
	const commonProps = {
		id: 'id',
		options: [
			{ text: 'Option 1', value: '1' },
			{ text: 'Option 2', value: '2' },
		],
		onChange: () => {},
	};
	if (type === 'select') {
		return <ChoiceList {...commonProps} type={type} label="choice" defaultValue={selected} />;
	} else {
		return <ChoiceList {...commonProps} type={type} legend="choice" defaultValue={selected} />;
	}
}

function createChoiceListWithExpansion(type: 'select' | 'radio' | 'check', selected?: string[]) {
	const commonProps = {
		id: 'id',
		options: [
			{ text: 'Option 1', value: '1' },
			{ text: 'Option 2', value: '2' },
		],
		onChange: () => {},
	};
	const expansion = (
		<ChoiceListExpansion showOnValue="1">
			<p>My expansion</p>
		</ChoiceListExpansion>
	);
	if (type === 'select') {
		return (
			<ChoiceList {...commonProps} type={type} label="choice" defaultValue={selected}>
				{expansion}
			</ChoiceList>
		);
	} else {
		return (
			<ChoiceList {...commonProps} type={type} legend="choice" defaultValue={selected}>
				{expansion}
			</ChoiceList>
		);
	}
}

describe('select', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceList('select'));
		expect(component).toMatchSnapshot();
	});

	it('selected prop is the selected input', () => {
		const component = render(createChoiceList('select', ['1']));
		expect(component.getByDisplayValue(/Option 1/i)).toBeTruthy();
	});
});

describe('select with expansion', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceListWithExpansion('select', ['1']));
		expect(component).toMatchSnapshot();
	});

	it('expansion is showed when corresponding input selected', () => {
		const component = render(createChoiceListWithExpansion('select', ['1']));
		expect(component.getByText(/expansion/i)).toBeTruthy();
	});
});

describe('check', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceList('check'));
		expect(component).toMatchSnapshot();
	});

	it('selected prop is the selected input', () => {
		const component = render(createChoiceList('check', ['1']));
		expect((component.getByDisplayValue(/1/i) as HTMLInputElement).checked).toBeTruthy();
	});
});

describe('select with expansion', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceListWithExpansion('check', ['1']));
		expect(component).toMatchSnapshot();
	});

	it('expansion is showed when corresponding input selected', () => {
		const component = render(createChoiceListWithExpansion('check', ['1']));
		expect(component.getByText(/expansion/i)).toBeTruthy();
	});
});

describe('radio', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceList('radio'));
		expect(component).toMatchSnapshot();
	});

	it('selected prop is the selected input', () => {
		const component = render(createChoiceList('radio', ['1']));
		expect((component.getByDisplayValue(/1/i) as HTMLInputElement).checked).toBeTruthy();
	});
});

describe('radio with expansion', () => {
	it('matches snapshot', () => {
		const component = render(createChoiceListWithExpansion('radio', ['1']));
		expect(component).toMatchSnapshot();
	});

	it('expansion is showed when corresponding input selected', () => {
		const component = render(createChoiceListWithExpansion('radio', ['1']));
		expect(component.getByText(/expansion/i)).toBeTruthy();
	});
});
