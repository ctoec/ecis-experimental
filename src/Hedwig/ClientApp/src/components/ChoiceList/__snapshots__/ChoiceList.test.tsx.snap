// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        aria-required="true"
        class="grid-gap grid-row usa-fieldset
			
			 undefined
			"
        id="id-fieldset"
      >
        <legend
          class="usa-sr-only"
          id="fieldset-legend-id-fieldset"
        >
          choice
        </legend>
        <div
          class="grid-gap grid-row margin-top-3"
        >
          <div
            class=""
          >
            <div
              class="usa-checkbox undefined"
            >
              <input
                class="usa-checkbox__input"
                id="id-1"
                name=""
                type="checkbox"
                value="1"
              />
              <label
                class="usa-checkbox__label"
                for="id-1"
              >
                Option 1
              </label>
            </div>
            <div
              class="usa-checkbox undefined"
            >
              <input
                class="usa-checkbox__input"
                id="id-2"
                name=""
                type="checkbox"
                value="2"
              />
              <label
                class="usa-checkbox__label"
                for="id-2"
              >
                Option 2
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset
      aria-required="true"
      class="grid-gap grid-row usa-fieldset
			
			 undefined
			"
      id="id-fieldset"
    >
      <legend
        class="usa-sr-only"
        id="fieldset-legend-id-fieldset"
      >
        choice
      </legend>
      <div
        class="grid-gap grid-row margin-top-3"
      >
        <div
          class=""
        >
          <div
            class="usa-checkbox undefined"
          >
            <input
              class="usa-checkbox__input"
              id="id-1"
              name=""
              type="checkbox"
              value="1"
            />
            <label
              class="usa-checkbox__label"
              for="id-1"
            >
              Option 1
            </label>
          </div>
          <div
            class="usa-checkbox undefined"
          >
            <input
              class="usa-checkbox__input"
              id="id-2"
              name=""
              type="checkbox"
              value="2"
            />
            <label
              class="usa-checkbox__label"
              for="id-2"
            >
              Option 2
            </label>
          </div>
        </div>
      </div>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`radio matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <fieldset
        aria-required="true"
        class="grid-gap grid-row usa-fieldset
			
			 undefined
			"
        id="id-fieldset"
      >
        <legend
          class="usa-sr-only"
          id="fieldset-legend-id-fieldset"
        >
          choice
        </legend>
        <div
          class="grid-gap grid-row margin-top-3"
        >
          <div
            class=""
          >
            <div
              class="usa-radio undefined"
            >
              <input
                class="usa-radio__input"
                id="1"
                name=""
                type="radio"
                value="1"
              />
              <label
                class="usa-radio__label"
                for="1"
              >
                Option 1
              </label>
            </div>
            <div
              class="usa-radio undefined"
            >
              <input
                class="usa-radio__input"
                id="2"
                name=""
                type="radio"
                value="2"
              />
              <label
                class="usa-radio__label"
                for="2"
              >
                Option 2
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </div>
  </body>,
  "container": <div>
    <fieldset
      aria-required="true"
      class="grid-gap grid-row usa-fieldset
			
			 undefined
			"
      id="id-fieldset"
    >
      <legend
        class="usa-sr-only"
        id="fieldset-legend-id-fieldset"
      >
        choice
      </legend>
      <div
        class="grid-gap grid-row margin-top-3"
      >
        <div
          class=""
        >
          <div
            class="usa-radio undefined"
          >
            <input
              class="usa-radio__input"
              id="1"
              name=""
              type="radio"
              value="1"
            />
            <label
              class="usa-radio__label"
              for="1"
            >
              Option 1
            </label>
          </div>
          <div
            class="usa-radio undefined"
          >
            <input
              class="usa-radio__input"
              id="2"
              name=""
              type="radio"
              value="2"
            />
            <label
              class="usa-radio__label"
              for="2"
            >
              Option 2
            </label>
          </div>
        </div>
      </div>
    </fieldset>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`select matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="usa-form-group "
      >
        <div
          class=""
        />
        <label
          class="usa-label"
          for="id"
        >
          choice
        </label>
        <select
          aria-required="true"
          class="usa-select"
          id="id"
          name="id"
        >
          <option
            value=""
          >
            - Select -
          </option>
          <option
            value="1"
          >
            Option 1
          </option>
          <option
            value="2"
          >
            Option 2
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="usa-form-group "
    >
      <div
        class=""
      />
      <label
        class="usa-label"
        for="id"
      >
        choice
      </label>
      <select
        aria-required="true"
        class="usa-select"
        id="id"
        name="id"
      >
        <option
          value=""
        >
          - Select -
        </option>
        <option
          value="1"
        >
          Option 1
        </option>
        <option
          value="2"
        >
          Option 2
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
